automation/tables.py

```diff
@@ -1,15 +1,126 @@
-from . import models
-from .functions import inbound, members
-from .clients import airtable
-
-INBOUND = airtable.TableSpec(
-    name="inbound",
-    model_cls=models.InboundModel,
-    status_to_cb={None: inbound.NewCallback},
-)
+import abc
+from functools import cached_property
+
+import sendgrid
+
+from . import config, models
+from .functions import delivery, inbound, members
+from .clients import airtable, slack
+
+
+class PollableTable(abc.ABC):
```

Comment:
This is mostly just refactoring how polling dispatch works. Now, we associate a `TableSpec` with how it handles status updates for polling by defining a `PollableTable` subclass that implements `on_status_update`. If you just want to query airtable, you can use `get_airtable` with a config, if you want to poll, construct an instance of the `PollableTable` and call `poll_table`.